x-pack/apm-server/sampling/pubsub/pubsub_test.go

```diff
@@ -0,0 +1,261 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pubsub_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/elastic/go-elasticsearch/v7"
+
+	"github.com/elastic/apm-server/x-pack/apm-server/sampling/pubsub"
+)
+
+func TestPublishSampledTraceIDs(t *testing.T) {
+	const (
+		indexName = "trace-ids"
+		beatID    = "beat_id"
+	)
+
+	requests := make(chan *http.Request, 1)
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var buf bytes.Buffer
+		if _, err := io.Copy(&buf, r.Body); err != nil {
+			panic(err)
+		}
+		r.Body = ioutil.NopCloser(&buf)
+
+		select {
+		case <-r.Context().Done():
+		case requests <- r:
+		}
+	}))
+	defer srv.Close()
+
+	client, err := elasticsearch.NewClient(elasticsearch.Config{
+		Addresses: []string{srv.URL},
+	})
+	require.NoError(t, err)
+
+	pub, err := pubsub.New(pubsub.Config{
+		Client:         client,
+		Index:          indexName,
+		BeatID:         beatID,
+		FlushInterval:  time.Millisecond,
+		SearchInterval: time.Minute,
+	})
+	require.NoError(t, err)
+
+	var ids []string
+	for i := 0; i < 20; i++ {
+		ids = append(ids, fmt.Sprintf("trace_%d", len(ids)))
```

Comment:
nit: since you are using ES built in `_seq_no` - would you mind using a random number for the trace IDs instead of a sequence just to make it clear that the value of the trace ID is not important for this.